web_programming/reddit_crawler_hot_topics.py

```diff
@@ -0,0 +1,33 @@
+import praw
+
+# creating an instance for praw reddit
+# get all the user cridentials form the reddit page: https://www.reddit.com/prefs/apps
+reddit = praw.Reddit(
+    # Client ID
+    client_id="Enter client id here",
+    # Client Secret Key
+    client_secret="Enter client secret here",
+    # User Agent
+    user_agent="Enter user agent here",
+    check_for_async=False
+)
+
+
+def crawler(subreddit: str) -> list[str]:
+    """
+    >>> crawler("all")
+    ["abcd","hijk","lmn","op","qr"]
+    """
+    data = []
+    hot = reddit.subreddit(subreddit).hot(limit=5)
+    for post in hot:
+        data.append(post.title)
+    return data

```

Comment:
Please replace with list comprehension as discussed in CONTRIBUTING.md.

Reply:
Sure, I shall make the changes as per suggestion